packages/contracts-periphery/contracts/universal/op-nft/CitizenshipBadge.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "@rari-capital/solmate/src/tokens/ERC721.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "./CitizenshipChecker.sol";
+import "./SocialContract.sol";
+
+contract CitizenshipBadge is ERC721 {
+    address public admin;
+
+    SocialContract public socialContract;
+    CitizenshipChecker public citizenshipChecker;
+
+    constructor(
+        address _admin,
+        address _socialContract,
+        address _citizenChecker
+    ) ERC721("Citizenship", "CB") {
+        admin = _admin;
+        socialContract = SocialContract(_socialContract);
+        citizenshipChecker = CitizenshipChecker(_citizenChecker);
+    }
+
+    function mint(address _to, bytes memory _proof) public returns (uint256) {
+        if (!citizenshipChecker.isCitizen(_to, _proof)) {
+            revert("CitizenshipBadge::mint: NOT_CITIZEN");
+        }
+        uint256 newItemId = uint256(uint160(_to));
+        _safeMint(_to, newItemId);
+        return newItemId;
+    }
+
+    function transferFrom(
+        address,
+        address,
+        uint256
+    ) public virtual override {
+        revert("CitizenshipBadge::transferFrom: SOUL_BOUND");
+    }
+
```

Comment:
You need to override `transfer` as well right?

Reply:
No, [solmate](https://github.com/transmissions11/solmate/blob/main/src/tokens/ERC721.sol) doesn't implement `transfer` and instead relies on `transferFrom`.

Reply:
should we leave comment about that both here and by the solmate import?